util/src/main/java/io/camunda/zeebe/util/allocation/DirectBufferAllocator.java

```diff
@@ -26,6 +27,7 @@ public AllocatedBuffer allocate(final int capacity) {
 
   private static void onFree(final AllocatedDirectBuffer buffer) {
     ALLOCATED_MEMORY.addAndGet(-buffer.capacity());
+    BufferUtil.free(buffer.rawBuffer);
```

Comment:
:thought_balloon: We need to discuss this whether we want to use it here again. I think we should free our resources, but a while ago we removed this (because we don't wanted to maintain the hacky free method). This is now supported by agrona.

Reply:
I think the worry was also how could we guarantee that the dispatcher was never accessed again after a call to free, since that would cause a segfault...except we later found out that that's not quite true, and calling free doesn't immediately free the memory :smile: You can try it yourself, allocate it, free it, then try to write :smile: It may be JVM dependent ofc, I don't know. I think the benefits outweigh the gains here :+1: